KAFKA-16558 Implemented HeartbeatRequestState toStringBase() and added a test for it

brenden20 opened the pull request:
I implemented HeartbeatRequestState.toStringBase(), I also added 2 getter methods for testing purposes. I added a test for the new method, which is passing. Test may/may not be necessary, added just in case.

brenden20 committed:
Added toStringBase

Added toStringBase

brenden20 committed:
Updated toStringBase() test

Added extra fields to stay consistent with toStringBase()

brenden20 committed:
Added reminder comment

Added reminder comment for hashCode() implementation

brenden20 committed:
Added getters and updated test

Added get methods for heartbeatTimer and heartbeatIntervalMs for testing purposes. Updated test to reflect changes

brenden20 committed:
Updated test

Updated toStringBase() test

brenden20 committed:
Removed unnecessary import

Removed unnecessary import

brenden20 committed:
Merge branch 'apache:trunk' into 16558

kirktrue commented on clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java:
```diff
@@ -152,6 +152,34 @@ public void cleanup() {
+        final long retryBackoffMs = 100;
+        final long retryBackoffMaxMs = 1000;
+        LogContext logContext = new LogContext();
+        HeartbeatRequestState heartbeatRequestState1 = new HeartbeatRequestState(
```
Super nit: we can drop the `1` at the end of the variable name, right?

```suggestion
        HeartbeatRequestState heartbeatRequestState = new HeartbeatRequestState(
```

kirktrue commented on clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java:
```diff
@@ -152,6 +152,34 @@ public void cleanup() {
+
+        RequestState requestState = new RequestState(
+                logContext,
+                "org.apache.kafka.clients.consumer.internals.HeartbeatRequestManager$HeartbeatRequestState",
```
Perhaps we could make `HeartbeatRequestState` package-protected, then we could do this:

```suggestion
                HeartbeatRequestManager.HeartbeatRequestState.class.getName(),
```

kirktrue commented on clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java:
```diff
@@ -478,6 +478,23 @@ public void resetTimer() {
+
+        // Visible for testing
+        protected Timer heartbeatTimer() {
+            return this.heartbeatTimer;
```
```suggestion
            return heartbeatTimer;
```

kirktrue commented on clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java:
```diff
@@ -478,6 +478,23 @@ public void resetTimer() {
+
+        // Visible for testing
+        protected long heartbeatIntervalMs() {
+            return this.heartbeatIntervalMs;
```
```suggestion
            return heartbeatIntervalMs;
```

kirktrue reviewed (changes requested):
Thanks @brenden20! This looks good. Minor nits only.

kirktrue commented:
The test failures are unrelated, FWIW.

brenden20 committed:
Fix heartbeatRequestState variable name

Changed heartbeatRequestState1 to heartbeatRequestState

brenden20 committed:
Changed HeartBeatRequestState to package protected

Changed HeartBeatRequestState to package protected and modified testToStringBase() to have cleaner code

brenden20 committed:
Cleaning up 'this' references

Modified heartbeatTimer() and heartbeatIntervalMs() methods, removed 'this.' in each.

brenden20 commented:
@kirktrue thank you for the suggestions, I have implemented and pushed your suggestions. Let me know if everything looks good!

brenden20 committed:
Merge branch 'apache:trunk' into 16558

philipnee commented on clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java:
```diff
@@ -452,7 +452,7 @@ private void handleFatalFailure(Throwable error) {
      * that it holds are:
      */
-    static class HeartbeatRequestState extends RequestState {
+    protected static class HeartbeatRequestState extends RequestState {
```
this is for testing purpose right? add a // Visible for testing comment

philipnee commented on clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java:
```diff
@@ -152,6 +152,34 @@ public void cleanup() {
+        HeartbeatRequestState heartbeatRequestState = new HeartbeatRequestState(
+                logContext,
+                time,
+                10,
```
extract this out as a variable for testing later

philipnee commented on clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java:
```diff
@@ -152,6 +152,34 @@ public void cleanup() {
+        );
+
+        String target = requestState.toStringBase() +
+                ", heartbeatTimer=" + heartbeatRequestState.heartbeatTimer() +
```
can we just apply `time.timer(heartbeatIntervalMs)`

philipnee commented on clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java:
```diff
@@ -152,6 +152,34 @@ public void cleanup() {
+
+        String target = requestState.toStringBase() +
+                ", heartbeatTimer=" + heartbeatRequestState.heartbeatTimer() +
+                ", heartbeatIntervalMs=" + heartbeatRequestState.heartbeatIntervalMs();
```
Just test against `heartbeatIntervalMs`

philipnee commented on clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java:
```diff
@@ -483,6 +483,23 @@ public void resetTimer() {
+                    ", heartbeatIntervalMs=" + heartbeatIntervalMs;
+        }
+
+        // Visible for testing
```
i don't think we need the `heartbeatTimer()` and `heartbeatIntervalMs()` here.  see the comment below.

philipnee commented on clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java:
```diff
@@ -152,6 +152,34 @@ public void cleanup() {
 
+    @Test
+    public void testHeartBeatRequestStateToStringBase() {
+        final long retryBackoffMs = 100;
```
i think we should try to consistently apply final to all variables if we want to use final.

philipnee commented:
hi @brenden20 - left some comments. lmk what do you think.

brenden20 committed:
Applied some suggestions

Applied some suggestions, consistent syntax and testing comment

brenden20 committed:
Updated test and removed get method

brenden20 committed:
Remove debug print

brenden20 committed:
Merge branch 'apache:trunk' into 16558

brenden20 committed:
Revert "Merge branch 'apache:trunk' into 16558"

This reverts commit 967b0f945859241b403b2c7c4f87f26c00d47962, reversing
changes made to d505968496c213bcc96f0fc5d941135a2d98cc48.

kirktrue reviewed (approved):
LGTM!

kirktrue commented:
@brenden20—can we mark this _Ready for review_?

brenden20 commented:
Marked ready to review

philipnee reviewed (approved):
lgtm thanks for the patch

brenden20 committed:
Merge branch 'apache:trunk' into 16558

brenden20 committed:
Merge branch 'apache:trunk' into 16558

brenden20 committed:
Merge branch 'apache:trunk' into 16558

brenden20 committed:
Merge branch 'apache:trunk' into 16558

brenden20 committed:
Merge branch 'apache:trunk' into 16558

brenden20 committed:
Merge branch '16558' of https://github.com/brenden20/kafka into 16558

brenden20 committed:
Revert "Merge branch '16558' of https://github.com/brenden20/kafka into 16558"

This reverts commit b9a85672cb8701f2233bf417de5a070843320b13, reversing
changes made to 64375adbf1d4242a14c91976768f800ec5a1eddb.

brenden20 committed:
Reapply "Merge branch '16558' of https://github.com/brenden20/kafka into 16558"

This reverts commit 4dff30d564d40ccb87858fd4c84532b9cf2a2760.

brenden20 committed:
Revert "Merge branch '16558' of https://github.com/brenden20/kafka into 16558"

This reverts commit b9a85672cb8701f2233bf417de5a070843320b13, reversing
changes made to 64375adbf1d4242a14c91976768f800ec5a1eddb.

brenden20 committed:
Fixing merge

brenden20 commented:
Merge issues, moved PR to https://github.com/apache/kafka/pull/16373 and closing this one
